From ff4edd03e75bfb995688e79d96fe253608f59b23 Mon Sep 17 00:00:00 2001
From: Amelie Delaunay <amelie.delaunay@foss.st.com>
Date: Fri, 31 Jan 2025 15:59:55 +0100
Subject: [PATCH 1499/1695] drivers: stm32_dbgmcu_mbx: manage secure debug
 request

At probe or on interrupt, when DBGMCU_DBG_AUTH_ACK.AUTH_HOST is set, it
means the debugger has written something in the mailbox, likely a
dbg_auth_start message, to start debug authentication protocol.
Call then psa_adac_start_secure_debug(), to initiate secure debug
connection and perform the secure debug authentication process.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 9363ad2df0e13f5c687f39a76a0949d0c5732adb drivers: stm32_dbgmcu_mbx: manage secure debug request]
Signed-off-by: Amelie Delaunay <amelie.delaunay@foss.st.com>
Change-Id: I2f53bb51d23b1804d2ef863c2ae0e23c26509b3e
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/433597
Domain-Review: Yann GAUTIER <yann.gautier@foss.st.com>
Reviewed-by: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
---
 core/drivers/stm32_dbgmcu_mbx.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/core/drivers/stm32_dbgmcu_mbx.c b/core/drivers/stm32_dbgmcu_mbx.c
index 5ccd42ff8..b3e67b87c 100644
--- a/core/drivers/stm32_dbgmcu_mbx.c
+++ b/core/drivers/stm32_dbgmcu_mbx.c
@@ -10,6 +10,7 @@
 #include <kernel/interrupt.h>
 #include <kernel/notif.h>
 #include <libfdt.h>
+#include <psa_adac_platform.h>
 #include <stdlib_ext.h>
 
 /* DBGMCU mailbox offset register */
@@ -84,7 +85,7 @@ TEE_Result stm32_dbgmcu_mbx_write_auth_dev(uint32_t value, uint32_t timeout_ms)
 /* TODO Maybe to move to platform */
 static void stm32_dbgmcu_mbx_secure_debug(void)
 {
-	//TODO
+	psa_adac_start_secure_debug();
 }
 
 static void stm32_dbgmcu_mbx_enable_interrupts(bool both)
-- 
2.47.2

